### PR TITLE
Add API Layer base

### PR DESCRIPTION
**Pull request by hulthe**
### Currently implemented endpoints:
  - /_matrix/client/r0/login
  - /_matrix/client/r0/sync
  - /_matrix/client/r0/publicRooms

### CODE REVIEW

File: src/main/java/com/github/cypher/sdk/api/Endpoint.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+	Endpoint(String s) {
+		name = "/_matrix/client/r0/".concat(s);
+	}
+
+	public boolean equalsName(String otherName) {
+		return name.equals(otherName);
+	}
+
+	public String toString() {
```
Comment (NotTrulyEvil) [inline src/main/java/com/github/cypher/sdk/api/Endpoint.java]: @Override missing

File: src/main/java/com/github/cypher/sdk/api/MatrixApiLayer.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+		JsonObject request  = new JsonObject();
+		request.addProperty("type", "m.login.password");
+		request.addProperty("user", username);
+		request.addProperty("password", password);
+
+		return Util.makeJsonPostRequest(url, request).getAsJsonObject();
+	}
+
+	public JsonObject sync(String filter, String since, boolean fullState, User.Presence setPresence) throws ExtendedHTTPException, IOException{
```
Comment (NotTrulyEvil) [inline src/main/java/com/github/cypher/sdk/api/MatrixApiLayer.java]: @Override tag missing

File: src/main/java/com/github/cypher/sdk/api/MatrixApiLayer.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		parameters.put("full_state"  , fullState ? "true" : "false");
+		parameters.put("access_token", accessToken);
+
+		URL url = Util.UrlBuilder(homeServer, Endpoint.SYNC, parameters);
+
+		JsonElement response = Util.makeJsonGetRequest(url);
+		return response.getAsJsonObject();
+	}
+
+	public JsonObject publicRooms(String server) throws ExtendedHTTPException, IOException {
```
Comment (NotTrulyEvil) [inline src/main/java/com/github/cypher/sdk/api/MatrixApiLayer.java]: @Override tag missing

**Comment by Gurgy** [comment]
ExtendedHTTPException could maybe be named RestHTTPException

**Pull request merged**